test/e2e/k8s_util.go

```diff
@@ -38,12 +39,14 @@ import (
 type KubernetesUtils struct {
 	*TestData
 	podCache map[string][]v1.Pod
+	podLock  *sync.Mutex
```

Comment:
The lock is not intended to be shared, it's more common to use `sync.Mutex` without the pointer, in which case you don't have to new it.

Reply:
done